src/test/regress/citus_tests/run_test.py

```diff
@@ -416,7 +422,23 @@ def find_test_schedule_and_line(test_name, args):
 
 def test_dependencies(test_name, test_schedule, schedule_line, args):
     if test_name in DEPS:
-        return DEPS[test_name]
+        # since enable_ddl_propagation is a must to execute tests
+        # below block adds enable_ddl_propagation as a dependency
+        # as the first element of extra tests if schedule is not configured
+        test_deps = DEPS[test_name]
+        ddl_propagation_test = (
+            "isolation_enable_ddl_propagation"
+            if test_name.startswith("isolation")
+            else "enable_ddl_propagation"
+        )
+
+        if test_deps is not None and test_deps.schedule is None:
+            test_deps.direct_extra_tests = [
+                ddl_propagation_test
+            ] + test_deps.direct_extra_tests
+            return test_deps
+        else:
+            return DEPS[test_name]
```

Comment:
Given that the base schedules already have enable_ddl_propagation and this function returns the appropriate base schedule for the test_name, do we need to explicitly add enable_ddl_propagation here?

Reply:
This function adds enable_ddl_propagation test just for the test items without schedule.

Reply:
check-style job on Github Actions anyway enforces each test to be utilized by at least one schedule, so I believe we should not need this.

Reply:
For enable_ddl_propagation, adding it in a schedule once is not enough. 
1. It should be defined at the top of each schedule to make sure that enable_ddl_propagation enabled from the very beginning
2. It should be defined at the top of the  dependencies list of the test items to make sure that test executes in enable_ddl_propagation

This line performs the second one. If I remove this line, I need to add enable_ddl_propagation test into dependencies list one by one. Additionally,  in the future if someone need to add a test without a schedule and with a dependency list, they need to add enable_ddl_propagation, which is likely to be forgotten

Reply:
Oh okay I got it now, thank you for explaining further.